### PR TITLE
remove kv stores from install.py

### PR DESCRIPTION
**Pull request by minrk**
As discussed in #155, I don't think we should be handling installation of kv stores. I think it's _very_ rare that installing a kv store is likely to be done via `install.py`, as 

- ~everywhere `install.py` is useful, a file provider is probably better,
- ~everywhere a kv store is desirable, an official container image is more likely how these will be installed, not in the same container as the Hub
- installing go binaries isn't complex for our target users, and both tools have simple copy/paste installs

### CODE REVIEW

File: docs/source/install.md [hunk 1]
```diff
@@ -14,41 +14,35 @@
...
+3. In order to be able to launch JupyterHub with traefik-proxy or run the tests, **traefik**, must first be installed and added to your `PATH`.
 
-   There are two ways you can install traefik, etcd and consul:
+   There are two ways you can install traefik:
 
    1. Through traefik-proxy's **install utility**.
 
       ```
-      $ python3 -m jupyterhub_traefik_proxy.install --traefik --etcd --consul --output=/usr/local/bin
+      $ python3 -m jupyterhub_traefik_proxy.install  --output=/usr/local/bin
```
Comment (consideRatio) [inline docs/source/install.md:24]:
```suggestion
      $ python3 -m jupyterhub_traefik_proxy.install --output=/usr/local/bin
```

**Review by consideRatio — APPROVED** [review]
I think from a maintenance perspective, its better that this is decoupled. If for example etcd reports a critical vulnerability and requests everybody patch, suddenly we find ourself in need of making a release right? I'd like to avoid this situation and think its entirely reasonable that this is offloaded to the jupyterhub admin or user that provides a distribution.

I think this PR requires approval of at least one more person than me to be merged, and it has failing tests currently I see. But, the changes LGTM from what I can tell.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -26,6 +26,8 @@ on:
 env:
   ETCDCTL_API: "3"
   PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION: python
+  ETCD_VERSION: "3.5.0"
```
Comment (GeorgianaElena) [inline .github/workflows/test.yml:29]: It looks like they don't recommend 3.5.0 for production, so maybe it's worth testing against latest etcd version instead?

Screenshot 2023-02-22 at 10 58 54

Comment (consideRatio) [inline .github/workflows/test.yml:29]:
```suggestion
  ETCD_VERSION: "3.5.7"
```

Comment (consideRatio) [reply .github/workflows/test.yml]: 3.5.0 is old, but 3.5.7 was out (and 3.4.24) --- they seem to maintain 3.4 and 3.5 actively.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -65,8 +67,15 @@ jobs:
...
-          python -m jupyterhub_traefik_proxy.install --traefik --etcd --consul --output=./bin
+          python -m jupyterhub_traefik_proxy.install --output=./bin
+
           pip freeze
+      - name: Install etcd, consul
+        run: |
+          curl -L https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_linux_amd64.zip > consul.zip
+          unzip consul.zip -d ./bin consul
+          curl -L https://github.com/etcd-io/etcd/releases/download/v${ETCD_VERSION}/etcd-v${ETCD_VERSION}-linux-amd64.tar.gz   > etcd.tar.gz
+          tar -xzf etcd.tar.gz -C ./bin --strip-components=1 '*/etcd*'
```
Comment (consideRatio) [inline .github/workflows/test.yml:78]:
```suggestion
          tar -xzf etcd.tar.gz -C ./bin --strip-components=1 --wildcards '*/etcd*'
```

**Comment by consideRatio** [comment]
Also with 3.5.7 we keep hanging, so maybe trial 3.4.x now

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -26,6 +26,8 @@ on:
 env:
   ETCDCTL_API: "3"
   PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION: python
+  ETCD_VERSION: "3.5.7"
```
Comment (GeorgianaElena) [inline .github/workflows/test.yml:29]:
```suggestion
  ETCD_VERSION: "3.4.24"
```

**Comment by GeorgianaElena** [comment]
Downgrading etcd to 3.4.24 raised the following error in tests:
`etcdmain: error verifying flags, invalid value "3.4.24" for ETCD_VERSION: parse error. See 'etcd --help'.`

It looks like it's because etcd uses internally the ETCD_VERSION env var https://github.com/etcd-io/etcd/pull/11225 and it's a bool. So, the last commit renames it to ETCD_DOWNLOAD_VERSION.

Let's see if this also fixes the hanging part

**Comment by minrk** [comment]
Thanks!

**Pull request merged**